gzip.ml

```diff
@@ -74,16 +77,37 @@ let open_in_chan ic =
     in_crc = Int32.zero }
 
 let open_in filename =
-  let ic = Pervasives.open_in_bin filename in
-  try
-    open_in_chan ic
-  with exn ->
-    Pervasives.close_in ic; raise exn
+  open_in_chan (Pervasives.open_in_bin filename)
```

Comment:
Note: this part of the diff, which reverts a part of 0e3fa45eeaf9f5c4d75e57d2ac1542821feea094, is a bad merge conflict resolution.